packages/connector/src/services/core/controllers/stream.ts

```diff
@@ -1,39 +1,46 @@
-import * as assert from 'assert'
-import { StreamServer } from '@interledger/stream-receiver'
 import { isIlpReply } from 'ilp-packet'
 import { RafikiContext, RafikiMiddleware } from '../rafiki'
 
-interface StreamControllerOptions {
-  serverSecret: Buffer
-  serverAddress: string
-}
+const CONNECTION_EXPIRY = 60 * 60 // seconds
 
-export function createStreamController({
-  serverSecret,
-  serverAddress
-}: StreamControllerOptions): RafikiMiddleware {
-  assert.equal(serverSecret.length, 32)
-  const server = new StreamServer({ serverSecret, serverAddress })
+// Track the total amount received per stream connection.
+export const streamReceivedKey = (connectionId: string): string =>
+  `stream_received:${connectionId}`
 
+export function createStreamController(): RafikiMiddleware {
   return async function (
     ctx: RafikiContext,
     next: () => Promise<unknown>
   ): Promise<void> {
+    const { redis, streamServer } = ctx.services
     const { request, response } = ctx
 
     const { stream } = ctx.accounts.outgoing
     if (
       !stream ||
       !stream.enabled ||
-      !server.decodePaymentTag(request.prepare.destination)
+      !streamServer.decodePaymentTag(request.prepare.destination) // XXX mark this earlier in the middleware pipeline
     ) {
       await next()
       return
     }
 
-    const moneyOrReply = server.createReply(request.prepare)
-    response.reply = isIlpReply(moneyOrReply)
-      ? moneyOrReply
-      : moneyOrReply.accept()
+    const moneyOrReply = streamServer.createReply(request.prepare)
+    if (isIlpReply(moneyOrReply)) {
+      response.reply = moneyOrReply
+      return
+    }
+
+    const { connectionId } = moneyOrReply
+    const totalReceived = await redis.incrby(
+      streamReceivedKey(connectionId),
+      +request.prepare.amount
```

Comment:
Should this be capped to `MAX_SAFE_INTEGER` so it's not coerced to a greater amount? Or, should we pass it as a string and overwrite the types?

I tested that `ioredis` _does_ support strings here, but the types don't reflect that. There's also a `stringNumbers` config option that can be passed to the constructor to return numbers as strings, but I'm guessing that probably also conflicts with the types.

Reply:
Or the connector could enforce a max packet amount of `MAX_SAFE_INTEGER`.